src/components/UriViewer/UriDownloadButton.js

```diff
@@ -55,15 +57,17 @@ export const UriDownloadButton = ({ uri, metadata: { bucket, name, fileName, siz
   useOnMount(() => {
     getUrl()
   })
+
+  const workspace = isAzureUri(url) ? azureWorkspace : extractWorkspaceDetails(workspaceStore.get().workspace)
```

Comment:
An Azure URL does not necessarily mean that the workspace is on Azure. A GCP workspace could contain a data table with references to files in Azure storage.

Reply:
I am really sorry I was misunderstanding your comments - let me know if the assumptions below are correct. 

1. For both workspaces, it would be better to pass the workspace prop coming from UriViewer to the button instead of grabbing it from global state as we do today for GCP
2. The workspace that we want to pass here is the workspace that the file is being downloaded from (not necessarily the workspace where the file may be) 

made appropriate changes based on these assumptions.

Reply:
1. Yes, we'd like the structure of Mixpanel events to be the same regardless of whether the current workspace is in GCP or Azure. So preferably using the `extractWorkspaceDetails` function for both.
2. Yes, for Mixpanel, we'd like to record the workspace that the user is currently working in.